### PR TITLE
[GH-371] Fixed issue: Getting multiple DMs when Oauth token is revoked

### LINKED ISSUE #371: DMs to inform the user their oAuth token has been revoked are received 3 or 4 times
When a user's GitLab oAuth token is revoked, the next time the plugin tries to fetch the counts a DM informs the user that they need to reconnect. 

This DM is received 3 or 4 times   

image

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
#### Summary
- The issue was caused because of the multiple API calls inside goroutines.
- Created a single API to get the data for LHS and todo command.

#### Screenshots
image

#### Possible risks:
- The changes in this PR might break once PR #394 is merged because of the variable/function name changes in that PR.

#### What to test?
- Getting a single DM when we try to refresh the count in the left-hand sidebar after the OAuth token is revoked from GitLab.
- We are getting proper data in the RHS.

###### Steps to reproduce:
1. Connect your GItlab account.
2. Revoke the OAuth token for your app from the [application page](https://gitlab.com/-/profile/applications) on Gitlab.
3. Refresh the counts in the left-hand sidebar.
image

###### Environment:
MM version: v7.8.2
Node version: 14.18.0
Go version: 1.19.0

#### Ticket Link
Fixes #371

### CODE REVIEW

**Review by mickmister — APPROVED** [review]
LGTM. Great improvement for consolidating LHS data into one API request as well

One thing that comes to mind is the behavior when there are multiple tabs of Mattermost open. For instance, the desktop app his 3 instances of the Mattermost webapp running at once.

**Comment by raghavaggarwal2308** [comment]
> One thing that comes to mind is the behavior when there are multiple tabs of Mattermost open. For instance, the desktop app his 3 instances of the Mattermost webapp running at once.

@mickmister  I am not sure what you are asking here. Can you please add some details?

File: webapp/src/components/sidebar_buttons/sidebar_buttons.jsx [hunk 1]
```diff
@@ -59,12 +56,7 @@ export default class SidebarButtons extends React.PureComponent {
         }
 
         this.setState({refreshing: true});
-        await Promise.all([
-            this.props.actions.getReviews(),
-            this.props.actions.getUnreads(),
-            this.props.actions.getYourPrs(),
-            this.props.actions.getYourAssignments(),
-        ]);
+        await this.props.actions.getLHSData();
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/components/sidebar_buttons/sidebar_buttons.jsx:59]: @mickmister In the left-hand sidebar we will not have any place to display the error. I think the only place we will be able to display the error will be the RHS. Currently, we are just logging the error in the server logs. Do you have any suggestions on it?

[the remark this answers is not in the document]
Comment (mickmister) [reply webapp/src/components/sidebar_buttons/sidebar_buttons.jsx:59]: The main thing I noticed is that we call `this.setState({refreshing: false})`  here, which will not run if an error happens. Even if it's as simple as logging the error, I think we should have some sort of error handling there to make sure we still clean up the state after the call.

**Comment by mickmister** [comment]
@raghavaggarwal2308 When multiple tabs of Mattermost are open, they all receive the same websocket messages. It's possible two browsers call the LHS data endpoint at the same time because of the websocket message syncing. It's probably unlikely for them to be that simultaneous, enough to not take it into account, but it's still worth bringing up for discussion since we're changing how the LHS data works in the PR

**Comment by raghavaggarwal2308** [comment]
@mickmister I have tested what you are saying for the following cases:
- When a user reconnects his account.
- When a webhook request is sent from Gitlab.
- When a user clicks on the screen after a certain period of time: In this case if multiple tabs are open, then only the tab the user had clicked on will be updated and on the remaining tabs, the user has to refresh manually. I think that's how it should work ideally. Please correct me if I am wrong. 

Everything was working fine. So, I don't think it will be an issue.

**Review by AayushChaudhary0001 — APPROVED** [review]
Tested and approved, working fine for single Dm and correct data shown in RHS, LGTM.

**Pull request merged**